### DISCUSSION TITLE
Is static analyzers broken?

### DISCUSSION THREAD

**Discussion by mnsgs** [TheLartians/Greeter#103] [Q&A] [answered]
Hi,

I am new to this template, and struggle to get the static analyzers working:

```
$ cmake -Hstandalone -DUSE_STATIC_ANALYZER=cppcheck -Bbuild/standalone
-- The CXX compiler identification is GNU 9.3.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Downloading CPM.cmake to /tmp/ModernCppStarter/build/standalone/cmake/CPM_0.31.1.cmake
-- CPM: adding package cmake-scripts@0 (3d2d5a9fb26f0ce24e3e4eaeeff686ec2ecfb3fb)
CMake Error at /tmp/ModernCppStarter/cmake/tools.cmake:55 (include):
  include could not find load file:

    /tools.cmake
Call Stack (most recent call first):
  CMakeLists.txt:7 (include)


CMake Error at /tmp/ModernCppStarter/cmake/tools.cmake:57 (clang_tidy):
  Unknown CMake command "clang_tidy".
Call Stack (most recent call first):
  CMakeLists.txt:7 (include)


-- Configuring incomplete, errors occurred!
See also "/tmp/ModernCppStarter/build/standalone/CMakeFiles/CMakeOutput.log".
```

Macro seems cloned and available:

```
$ rg --no-ignore-vcs clang_tidy
build/standalone/_deps/cmake-scripts-src/tools.cmake
22:macro(clang_tidy)

build/standalone/_deps/cmake-scripts-src/README.md
264:To use, add the `clang_tidy()` function, with the arguments being the options to pass to the clang tidy program, such as '-checks=*'.

build/standalone/_deps/cmake-scripts-src/example/all/CMakeLists.txt
24:clang_tidy(-format-style=file -checks=* -header-filter='${CMAKE_SOURCE_DIR}/*')

cmake/tools.cmake
57:    clang_tidy(${CLANG_TIDY_ARGS})
```

Output is the same for other analyzers - am I holding it wrong?

Version is 
```
$ git describe --tags 
v0.17.0-3-g4212173
```

Thanks,
Martin

**Comment by ClausKlein** [comment]
Hi Martin, yes may be, but IMHO if you use external packages it may not a good idea to check the entire project.

see https://github.com/TheLartians/ModernCppStarter/issues/89

**Reply by mnsgs** [reply]
Hi @ClausKlein, so we have two issues:

1. Analyzers triggered through ModernCppStarter is not working
2. https://discourse.cmake.org/t/how-to-prevent-clang-tidy-to-check-sources-added-with-fetchcontents/2871

I would assume 2. to be out of scope for a fix here, but also that 1. belongs somewhere between https://github.com/TheLartians/ModernCppStarter and https://github.com/StableCoder/cmake-scripts

Do you agree?

**Reply by ClausKlein** [reply]
Yes, but normally people  will not want to check the extern `fmt `or the `doctest` implementations!

So, if it works, it would checks mostly extern code.

**Reply by mnsgs** [reply]
> Yes, but normally people will not want to check the extern `fmt `or the `doctest` implementations!

Agree, and I assume this issue to be handled through https://discourse.cmake.org/t/how-to-prevent-clang-tidy-to-check-sources-added-with-fetchcontents/2871 . 

However, for 1., do you prefer me to create an issue here for that? At least that would highlight for others, that feature is currently broken.

**Comment by TheLartians** [comment, marked as answer]
Hey @mnsgs, thanks for raising the question! It does seem that we broke static analysers in a recent update to the new CPM.cmake shorthand syntax. Luckily the mistake has already been addressed in #111, so static analysers should be working again as expected!

**Reply by mnsgs** [reply]
Thanks for the headsup @TheLartians , I'll try to find time for retest - cheers :)

**Reply by TheLartians** [reply]
@mnsgs did the fix help solve your issue?

**Reply by mnsgs** [reply]
@TheLartians , sorry I haven't had the time to get back to this. I suggest to assume it is resolved for now. Thanks!

**Reply by TheLartians** [reply]
Great, thanks for letting me know!
